### PR TITLE
adding ability to add comments to config settings in main.cf

### PR DESCRIPTION
**Pull request by smbambling**

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -44,7 +44,15 @@
 
   case $ensure {
     'present': {
-      $changes = "set ${name} '${value}'"
+      if $comment {
+        $changes = [
+          "set ${name} '${value}'",
+          "insert #comment before $name",
```
Comment (raphink) [inline manifests/config.pp]: This is not idempotent. This code will insert a comment every time Puppet runs.

Comment (smbambling) [inline manifests/config.pp]: Hmm...I'll reset this soon as I get a chance.  If you would like I can cancel this PR until further testing.

Comment (raphink) [inline manifests/config.pp]: You can keep the PR and just push force to your branch, I'm fine with that.

Comment (raphink) [inline manifests/config.pp]: I do think that managing this idempotently will probably require a type/provider. Are you familiar with augeasproviders.com?

**Comment by raphink** [comment]
This PR hasn't been updated in a very long time. I'm going to close it for now.

Please reopen if you wish to address the comments.

**Pull request closed without merging**